src/assets/TokensController.ts

```diff
@@ -498,73 +680,64 @@ export class TokensController extends BaseController<
     this.update({ suggestedAssets: [...newSuggestedAssets] });
   }
 
-  /**
-   * Removes a token from the stored token list and saves it in ignored tokens list.
-   *
-   * @param address - The hex address of the token contract.
-   */
-  removeAndIgnoreToken(address: string) {
-    address = toChecksumHexAddress(address);
-    const { tokens, ignoredTokens } = this.state;
-
-    const alreadyIgnored = ignoredTokens.find(
-      (tokenAddress) => tokenAddress.toLowerCase() === address.toLowerCase(),
-    );
-
-    const newTokens = tokens.filter((token) => {
-      if (token.address.toLowerCase() === address.toLowerCase()) {
-        !alreadyIgnored && ignoredTokens.push(address);
-        return false;
-      }
-      return true;
-    });
-
-    const { newAllTokens, newAllIgnoredTokens } = this._getNewAllTokensState(
-      newTokens,
-      ignoredTokens,
-    );
-
-    this.update({
-      allTokens: newAllTokens,
-      tokens: newTokens,
-      allIgnoredTokens: newAllIgnoredTokens,
-      ignoredTokens,
-    });
-  }
-
   /**
    * Takes a new tokens and ignoredTokens array for the current network/account combination
    * and returns new allTokens and allIgnoredTokens state to update to.
    *
-   * @param newTokens - The new tokens to set for the current network and selected account.
-   * @param newIgnoredTokens - The new ignored tokens to set for the current network and selected account.
+   * @param params - Object that holds token params.
+   * @param params.newTokens - The new tokens to set for the current network and selected account.
+   * @param params.newIgnoredTokens - The new ignored tokens to set for the current network and selected account.
+   * @param params.newDetectedTokens - The new detected tokens to set for the current network and selected account.
    * @returns The updated `allTokens` and `allIgnoredTokens` state.
    */
-  _getNewAllTokensState(newTokens: Token[], newIgnoredTokens: string[]) {
-    const { allTokens, allIgnoredTokens } = this.state;
+  _getNewAllTokensState(params: {
+    newTokens?: Token[];
+    newIgnoredTokens?: string[];
+    newDetectedTokens?: Token[];
+  }) {
+    const { newTokens, newIgnoredTokens, newDetectedTokens } = params;
+    const { allTokens, allIgnoredTokens, allDetectedTokens } = this.state;
     const { chainId, selectedAddress } = this.config;
-    const networkTokens = allTokens[chainId];
-    const networkIgnoredTokens = allIgnoredTokens[chainId];
 
-    const newNetworkTokens = {
-      ...networkTokens,
-      ...{ [selectedAddress]: newTokens },
-    };
-    const newIgnoredNetworkTokens = {
-      ...networkIgnoredTokens,
-      ...{ [selectedAddress]: newIgnoredTokens },
-    };
+    let newAllTokens = allTokens;
+    if (newTokens) {
```

Comment:
The `else` case is not covered here by any test. The same goes for the `if (newDetectedTokens) ` else case below